syntax = "proto2";

package GC_LS;

enum MsgID
{
	GC2LS_AskLogin = 100;
	LS2GC_NotifyLoginResult = 101;
	LS2GC_NotifyServerBSAddr = 102;
	LS2GC_Error = 103;
}

message AskLogin
{
	optional MsgID msgid = 1[default = GC2LS_AskLogin];
	optional uint32 platform = 2;
	optional string mail = 3;
	optional string password = 4;
}

message ServerInfo
{
	optional string ServerName = 1;
	optional string ServerAddr = 2;
	optional int32 ServerPort = 3;
	optional int32 ServerState = 4;
}

message ServerBSAddr
{
	optional MsgID msgid = 1 [default = LS2GC_NotifyServerBSAddr];
	repeated ServerInfo serverinfo = 2;
}

message ErrorMsg
{
	optional MsgID msgid = 1 [default = LS2GC_Error];
	optional int32 errorid = 2;
}
